### PR TITLE
Fix #1344 - show clinical assessment/dismiss on research page.

### LINKED ISSUE #1344: Dismissed variants for a case should show on research view
If a variant is dismissed on clinical analysis, it should show as such also on research..

### PR DESCRIPTION
**Pull request by dnil**
This PR adds a functionality - show clinical assessments / dismissals on research page.

Screenshot 2019-11-28 at 11 21 12

**How to test**:
1. manually rank, dismiss, tag, comment and evaluate to your hearts content on a clinical case variant page
1. ensure that you can see shadows from all your clinical assessments on the research page

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by dnil** [comment]
Ok, this now works, and it may be a good time to get some input. I'll try to find some time to write tests before it is really ready.

**Comment by dnil** [comment]
Yeah, well " Error: HTTP Error 503: Service Unavailable" on ENSEMBL. But at least the PR test passed - and importantly to you @northwestwitch, coveralls passes! 😉

**Comment by dnil** [comment]
Also fixes a bug where current master crashes on case page if a manually ranked variant is pinned for the case.

**Comment by northwestwitch** [comment]
Reviewing this now!

**Comment by northwestwitch** [comment]
> or: HTTP Error 503: Service Unavailable" on ENSEMBL. But at least the PR test passed - and importantly to you @northwestwitch, coveralls pa

**Comment by northwestwitch** [comment]
> Yeah, well " Error: HTTP Error 503: Service Unavailable" on ENSEMBL. But at least the PR test passed - and importantly to you @northwestwitch, coveralls passes! 😉

Sorry, I closed the branch by mistake. I think you might have been lucky and had travis testing while Ensembl was up for a short while. It has been working intermittently since Friday and in order to merge other PRs we had to skip those tests. Anyway I've reported the problem to EBI, let's hope they fix it soon!

File: scout/server/blueprints/variants/controllers.py [hunk 1]
```diff
@@ -49,28 +50,46 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
         overlapping_svs = [sv for sv in store.overlapping(variant_obj)]
         variant_obj['overlapping'] = overlapping_svs or None
 
+        # show previous classifications for research variants,
+        is_research = False
+        if variant_obj['variant_type'] == 'research':
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py]: What about:
`is_research = variant_obj['variant_type'] == 'research'`

File: scout/server/blueprints/variants/controllers.py [hunk 2]
```diff
@@ -49,28 +50,46 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
...
+            is_research = True
+
         # Get all previous ACMG evalautions of the variant
         evaluations = []
         for evaluation_obj in store.get_evaluations(variant_obj):
             classification = evaluation_obj['classification']
             # Only show pathogenic/likely pathogenic from other cases on variants page
             if evaluation_obj['case_id'] == case_obj['_id']:
-                continue
+                if not is_research:
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py]: Could this if be put before, so the loop doesn't even start?

File: scout/server/blueprints/variants/controllers.py [hunk 3]
```diff
@@ -49,28 +50,46 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
...
+                    continue
             if not classification in ['pathogenic', 'likely_pathogenic']:
-                continue
+                if not is_research:
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py]: So you avoid this one as well

File: scout/server/blueprints/variants/controllers.py [hunk 4]
```diff
@@ -49,28 +50,46 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
...
             # Convert the classification int to readable string
             evaluation_obj['classification'] = ACMG_COMPLETE_MAP.get(classification)
             evaluations.append(evaluation_obj)
         variant_obj['evaluations'] = evaluations
 
+        # Show previous classifications from the clinical side for research
+        if is_research:
+            # get variant by simple_id. That will really just return the first variant found -
+            # but mostly that would be clinical.. Its a start.
+            clinical_var_obj = store.variant(case_id=case_obj['_id'],
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py]: What about querying all events by simple_id ('subject')? That would give an indication of the actions associated to the specific clinical variants, and from there you could collect their _id or document_id

File: scout/server/blueprints/variants/controllers.py (hunk 2 above)
Comment (dnil) [reply scout/server/blueprints/variants/controllers.py]: Hm, not easily without making two loops? We still want to go through previous evals for both research and non research.

Comment (dnil) [reply scout/server/blueprints/variants/controllers.py]: I'll add in a a couple of comments that may make it more clear.

File: scout/server/blueprints/variants/controllers.py (hunk 4 above)
Comment (dnil) [reply scout/server/blueprints/variants/controllers.py]: Oh, sorry about the comment - I gave the adapter variant an extra argument. It is better this way: before it was a lottery for anyone which type of variant would be returned if both clinical and research existed.

File: scout/server/blueprints/variants/controllers.py [hunk 5]
```diff
@@ -73,8 +73,7 @@ def variants(store, institute_obj, case_obj, variants_query, page=1, per_page=50
 
         # Show previous classifications from the clinical side for research
         if is_research:
-            # get variant by simple_id. That will really just return the first variant found -
-            # but mostly that would be clinical.. Its a start.
+            # get variant by simple_id.
```
Comment (northwestwitch) [inline scout/server/blueprints/variants/controllers.py]: 👍

**Pull request merged**